### PR TITLE
Adds dextrous guardian

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
rscadd: Added dextrous guardians to the code, able to hold and use items and store a single item within themselves.
experiment: Dextrous guardians do low damage on punches, have medium damage resist, and recalling or leashing will force them to drop any items in their hands.
/:cl:

merge token; #19252

oh christ this took a lot of code to get working
drones now use a subtype of the dextrous hud
ghosts can once again open bags and boxes to look at them(did this cause a bug I don't know I sure hope the comment is accurate)
space station 13 2016-07-11 080759

_THIS IS SO UNWISE~_

### CODE REVIEW

**Comment by TechnoAlchemisto** [comment]
can they use guns

**Comment by ChangelingRain** [comment]
If a drone can use it, the guardian can use it.

File: code/game/objects/structures/beds_chairs/chair.dm [hunk 1]
```diff
@@ -204,7 +200,18 @@
 /obj/structure/chair/MouseDrop(over_object, src_location, over_location)
 	. = ..()
 	if(over_object == usr && Adjacent(usr))
-		if(!item_chair || !ishuman(usr) || has_buckled_mobs() || src.flags & NODECONSTRUCT)
+		var/can_hold_items = FALSE
```
Comment (RemieRichards) [inline code/game/objects/structures/beds_chairs/chair.dm]: this block smells like it needs to be a proc.

File: code/modules/mob/living/simple_animal/guardian/types/dextrous.dm [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	if(I.pulledby)
+		I.pulledby.stop_pulling()
+
+	I.screen_loc = null // will get moved if inventory is visible
+	I.loc = src
+	I.equipped(src, slot)
+	I.layer = ABOVE_HUD_LAYER
+
+	switch(slot)
+		if(slot_drone_storage)
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/guardian/types/dextrous.dm]: the naming scheme of these vars and the defines for where things go on the HUD probably shouldn't reference "drone" anymore, probably ```slot_dexterous_animal_storage``` or something, and the same for the others of course.

**Comment by Xhuis** [comment]
There's some fixes and such that I put in #19065 that didn't get in - make sure that those are in or this may not work.

**Comment by ChangelingRain** [comment]
Excluding throwing, it's got all the fixes it needs to work.

**Comment by KorPhaeron** [comment]
Stunproof creature being able to use a stun baton or taser sounds terrible

**Comment by ChangelingRain** [comment]
I still need to give it hand overlays and maybe do something about that, yeah.
I'm not sure what, exactly, I can do about it having stuns, melee weapons in general stun so it'd probably be difficult.

**Comment by Zilenan91** [comment]
Yeah Holos shouldn't be able to use any kind of stun. 

Can holos hold TTVs in them? It'll be foolproof, no surgery required bombing if they can, and undiscoverable without killing the person, searching wouldn't work.

**Comment by ChangelingRain** [comment]
Hand icon handling done, I really don't know what you'd do to prevent it from using stuns without just preventing all guns/stunbatons/bunches of other stunny stuff.

Oh, I know. I'll make it snapping back to its user force it to drop its items, so it's basically great utility but if it lags behind or the owner is dragged away it loses whatever it was holding.

**Comment by ChangelingRain** [comment]
Also yes they can hold TTVs but using the holo to bomb is still not very practical because it has less access than its owner does and is very conspicuous. Unless you mean suicide bombing in which case why are you using a holo to do that, cripes.

At best it'd be able to drop a bomb, recall, and then have the bomb signalled/signal it on its own.

**Comment by Zilenan91** [comment]
For suicide bombing you need to set up a voice code, and that runs the risk of someone accidentally saying it. For Holos if they can have hands they can store the bomb inside them and then hold the signaller. 

They don't drop things in their hands do they? If they don't then there's no way to stop this bomb.

**Comment by ChangelingRain** [comment]
They drop things in their hands when they recall or are forced to leash back to their summoner, yes.

**Comment by Zilenan91** [comment]
Nevermind then

**Comment by KorPhaeron** [comment]
Can they still use tasers

**Comment by ChangelingRain** [comment]
yes; would you like me to make them admin-edit only

**Comment by coiax** [comment]
At the risk of invoking the balance gods, guardians are currently only legitimately obtainable through tarot cards, and that's randomly selected. So if you have, what is basically, a loyal drone that shares your HP pool, you're only gonna get it if it's selected out of the like 12 other classes we have.

The only other comparison we have is Nukeop's Reinforcements, which costs 25TC (they come with free agent ID and stetchkin, worth 7 + 2TC, meaning the cost of the body is 16TC)

**Comment by KorPhaeron** [comment]
Why would you you compare nuke op prices to traitor prices though

**Comment by coiax** [comment]
they use the same cost pool, like 99% of the time.

**Pull request merged**